src/db/dal/variant.ts

```diff
@@ -1,35 +1,55 @@
 import { Op } from 'sequelize';
+
 import VariantModel from '../models/Variant';
 
-export const addNewEntry = async function (uniqueId: string, organizationId: string, authorId: string, properties: any) {
-    return await VariantModel.create({
-        unique_id: uniqueId,
-        organization_id: organizationId,
-        author_id: authorId,
-        properties,
-        timestamp: new Date()
+export const addNewEntry = async function (
+    uniqueId: string,
+    organizationId: string,
+    authorId: string,
+    properties: any,
+) {
+    return await VariantModel.findOne({
+        order: [['timestamp', 'DESC']],
+        where: {
+            unique_id: {
+                [Op.eq]: uniqueId,
+            },
+            author_id: {
+                [Op.eq]: authorId,
+            },
+        },
+    }).then(async function onResponse(value: VariantModel) {
```

Comment:
Si tu utilise le await pas besoin du then, je l'écrirais plutôt sous cette forme:
```suggestion
const variantFound = await VariantModel.findOne({
        order: [['timestamp', 'DESC']],
        where: {
            unique_id: {
                [Op.eq]: uniqueId,
            },
            author_id: {
                [Op.eq]: authorId,
            },
        },
    });

    const variantCreated = await VariantModel.create({
        unique_id: uniqueId,
        organization_id: organizationId,
        author_id: authorId,
        properties: {
            ...(variantFound?.properties || {}),
            ...properties,
        },
        timestamp: new Date(),
    });

    return variantCreated;
```